### PR TITLE
Example of how to unit test a controller method which supports odata

### PR DESCRIPTION
**Pull request by andrei-galkin**
Hello guys, 

This is the test project example how controllers can be tested with odata protocol support.

### CODE REVIEW

File: sample/ODataRoutingSample.Tests/AccountsControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            var modelBuilder = new ODataConventionModelBuilder();
+            modelBuilder.EntitySet<Account>("Account");
+            var edmModel = modelBuilder.GetEdmModel();
+
+            var options = new ODataOptions();
+            options.AddModel("odata", EdmCoreModel.Instance);
+            options.AddModel("my{data}", edmModel);
+
+            HttpRequest request = RequestFactory.Create("GET",
+                "http://localhost/api?$filter=AccountInfo/Id eq 2&$inlinecount=allpages",
```
Comment (xuzhg) [inline sample/ODataRoutingSample.Tests/AccountsControllerTests.cs:75]: $inlinecount=allpages?

Are you sure we test OData v3?

File: sample/ODataRoutingSample.Tests/AccountsControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ODataRoutingSample.Controllers;
+using ODataRoutingSample.Models;
+
+namespace ODataRoutingSample.Tests
+{
+    /// <summary>
+    /// This is a suggestion how to unit test a controller which will be very useful for the devs
+    /// </summary>
+    [TestClass]
```
Comment (xuzhg) [inline sample/ODataRoutingSample.Tests/AccountsControllerTests.cs:20]: I'd like to use xunit

**Comment by xuzhg** [comment]
@andrei-galkin why don't add similar test cases (xunit) into my e2e test project?

File: sample/ODataRoutingSample.Tests/AccountsControllerTests.cs (hunk 2 above)
Comment (andrei-galkin) [reply sample/ODataRoutingSample.Tests/AccountsControllerTests.cs:20]: I will update it.

File: sample/ODataRoutingSample.Tests/AccountsControllerTests.cs (hunk 1 above)
Comment (andrei-galkin) [reply sample/ODataRoutingSample.Tests/AccountsControllerTests.cs:75]: Hmm. Will check it.

**Comment by andrei-galkin** [comment]
@xuzhg I will follow your suggestion.

File: sample/ODataRoutingSample.Tests/AccountsControllerTests.cs (hunk 1 above)
Comment (andrei-galkin) [reply sample/ODataRoutingSample.Tests/AccountsControllerTests.cs]: Corrected

**Comment by andrei-galkin** [comment]
Hello @xuzhg, 
I corrected what was suggested. 
I check once again this suggestion - "why don't add similar test cases (xunit) into my e2e test project?"
My idea to give the real example example on the real project how to unit test a controller's method which supports ODATA. Personally, I spent day to figure it out and I would be glad to help developer who are doing the same. 

Thank you for the review
Andrei

**Comment by xuzhg** [comment]
@andrei-galkin We switched the default branch from master to main and this PR is accidentally closed. 
If you still want me to accept it, please:
1) Create a new PR using main branch

Thanks.

**Comment by andrei-galkin** [comment]
Thank you @xuzhg. Created https://github.com/OData/AspNetCoreOData/pull/438.

**Pull request closed without merging**